### PR TITLE
chore(api): support using personal tokes

### PR DESCRIPTION
**Pull request by appleboy**
See https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html

### What did you implement:

Support using personal access tokens

### How did you implement it:

Add new API function: `NewAPIWithToken`

### How can we verify it:

Create your API token and verify with the following example code:

```go
package main

import (
  "fmt"

  cf "github.com/essentialkaos/go-confluence/v5"
)

func main() {
  api, err := cf.NewAPIWithToken("https://confluence.domain.com", "your token")
  api.SetUserAgent("MyApp", "1.2.3")

  if err != nil {
    fmt.Printf("Error: %v\n", err)
    return
  }

  content, err := api.GetContentByID(
    "18173522", cf.ContentIDParameters{
      Version: 4,
      Expand:  []string{"space", "body.view", "version"},
    },
  )
  if err != nil {
    fmt.Printf("Error: %v\n", err)
    return
  }

  fmt.Printf("ID: %s\n", content.ID)
}
```

**Is this ready for review?:** Yes
**Is it a breaking change?:** No

### CODE REVIEW

File: confluence.go [hunk 1]
```diff
@@ -103,7 +127,6 @@ func (api *API) GetAuditRecords(params AuditParameters) (*AuditRecordCollection,
 		"GET", "/rest/api/audit",
 		params, result, nil,
 	)
-
```
Comment (andyone) [inline confluence.go:106]: Please do not change code formatting for other parts of code.

File: confluence.go [hunk 2]
```diff
@@ -88,6 +89,29 @@ func NewAPI(url, username, password string) (*API, error) {
 	}, nil
 }
 
+// NewAPIWithToken create new API struct using Token
+// See https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html
+// Support Confluence 7.9 and later
+func NewAPIWithToken(url, token string) (*API, error) {
+	switch {
```
Comment (andyone) [inline confluence.go:96]: You have to add a check and error for an empty token value. And maybe an additional check for token length (it must be 44 symbols long).

**Comment by andyone** [comment]
Hi! Thanks for your contribution!

**Review by andyone — CHANGES_REQUESTED** [review]

File: confluence.go (hunk 2 above)
Comment (appleboy) [reply confluence.go:98]: 3946d9c c062f9b

File: confluence.go (hunk 1 above)
Comment (appleboy) [reply confluence.go]: Done

**Comment by appleboy** [comment]
@andyone Please help to review again.

File: _example/main.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+
+	cf "github.com/essentialkaos/go-confluence/v5"
+)
+
+func main() {
```
Comment (andyone) [inline _example/main.go:9]: I'm not sure that it will be really helpful for users if we provide just one example with couple of methods.

Comment (andyone) [reply _example/main.go:9]: I plan to move these basic usage examples into `example_test.go`. In this case, these examples will be shown on [pkg.go.dev](https://pkg.go.dev/github.com/essentialkaos/go-confluence/v5). So, I can merge it like that and then refactor them as a part of preparing for release.

Comment (appleboy) [reply _example/main.go:9]: @andyone Sorry I will remove the example that is just for testing

Comment (appleboy) [reply _example/main.go]: [5fd41b4](https://github.com/essentialkaos/go-confluence/pull/52/commits/5fd41b488acd05aa840270b43395f7d38f72a929) Done

**Comment by andyone** [comment]
Looks good, merged. Thanks!

**Pull request merged**